### PR TITLE
fix(compiler): fix remove whitespaces from inside the element

### LINKED ISSUE #984: Fix bug that makes Compiler remove whitespaces from inside the element
Notice the leading space inside the `<span>`. Compiler removes it, breaking the formatting and removing extra space that is needed for text formatting.

```jsx
<h1 className="text-5xl md:text-7xl font-bold text-white mb-6 leading-tight">
  Hello World
  <span className="bg-gradient-to-r from-purple-400 via-pink-400 to-yellow-400 bg-clip-text text-transparent"> From Lingo.dev Compiler</span>
</h1>
```

TODO:
- [ ] compiler should not remove leading space from a component / tag if present (eg. inside the `<span>` above)
- [ ] it should still remove extra whitespace / lines (eg. before "Hello World" in `<h1>` above)

Investigate:
- can you use [dedent](https://www.npmjs.com/package/dedent) for this?

### PR DESCRIPTION
**Pull request by VAIBHAVSING**
fixes #984 
This pull request introduces improvements to whitespace normalization in JSX content handling within the Lingo.dev compiler. The key changes include refining the logic for preserving meaningful whitespace, adding targeted tests to ensure correctness, and cleaning up debug files. These updates enhance the robustness of JSX whitespace handling and improve code maintainability.

### Whitespace Normalization Enhancements:
* Improved the `normalizeJsxWhitespace` function to distinguish between formatting whitespace and meaningful leading/trailing spaces, ensuring proper preservation and collapsing of spaces as needed. Added handling for single-line and multi-line content separately. (`packages/compiler/src/utils/jsx-content.ts`, [packages/compiler/src/utils/jsx-content.tsR129-R184](diffhunk://#diff-c221deb69b53a09b21cfe5551de18cc0066b489bd26c7357e8005c3d9854e167R129-R184))

### Testing Additions:
* Added new tests in `jsx-content-whitespace.spec.ts` to verify the handling of leading spaces, explicit whitespace (`{" "}`), and nested JSX elements. These tests ensure the correctness of the updated `normalizeJsxWhitespace` logic. (`packages/compiler/src/utils/jsx-content-whitespace.spec.ts`, [packages/compiler/src/utils/jsx-content-whitespace.spec.tsR1-R67](diffhunk://#diff-f10d062be26cc2220ac9b518c885ca1e7afb591655c3dfb8b0a1e24ee8c4099eR1-R67))

### Documentation and Cleanup:
* Documented the changes in `.changeset/large-zoos-explain.md` and removed unnecessary debug/test files created during development. (`.changeset/large-zoos-explain.md`, [.changeset/large-zoos-explain.mdR1-R10](diffhunk://#diff-c5f46f64eb861f1527364228b4095753860bb876654999b06c88c5de1da4e5c0R1-R10))

### CODE REVIEW

File: packages/compiler/src/utils/jsx-content.ts [hunk 1]
```diff
@@ -126,25 +126,61 @@ function isWhitespace(nodePath: NodePath<t.JSXExpressionContainer>) {
 }
 
 function normalizeJsxWhitespace(input: string) {
```
Comment (mathio) [inline packages/compiler/src/utils/jsx-content.ts:128]: The logic here seems rather complex. I dont suppose there is a way to simplify this? I'm not saying this is wrong, I'm  just curious since we will need to maintain this piece of code 😅

**Comment by mathio** [comment]
Sorry for the force-push, i just rebased on latest `main` via GitHub UI.

File: packages/compiler/src/utils/jsx-content.ts (hunk 1 above)
Comment (VAIBHAVSING) [reply packages/compiler/src/utils/jsx-content.ts:128]: Thanks for pointing it out. I see that the function already exists here:
https://github.com/lingodotdev/lingo.dev/blob/30faa6d10e851a38ced86ae403b3a1fd48440bca/packages/compiler/src/utils/jsx-content.ts#L128

Let me know what exactly needs to be changed or improved.
Happy to update it accordingly.

Thank you!

**Comment by mathio** [comment]
Unit tests pass ✅  I tried a few other edge-cases manually in the function and it worked as well 👍 

---

I am testing manually in [next.js app demo](https://github.com/lingodotdev/lingo.dev/tree/main/demo/next-app) now.


1. I updated `hero-subtitle.tsx` to:
```jsx
export async function HeroSubtitle() {
  return (
    <p className="text-lg sm:text-xl text-gray-600 mb-10 max-w-xl mx-auto leading-relaxed">
      Localize your React app in every language in minutes. Scale to millions
      <b> from day one</b>.
    </p>
  );
}
```

2. There is a missing space in UI:
Screenshot 2025-07-03 at 19 29 02

4. As well as in source code:
Screenshot 2025-07-03 at 19 28 55

Please have a look. Maybe the issue is caused somewhere else?

**Comment by VAIBHAVSING** [comment]
I will take a look on it

**Review by mathio — CHANGES_REQUESTED** [review]
It is not ideal, but this functionality needs to be tested manually. You can do so using `demo/next-app` application and updating [one of the components like this](https://github.com/lingodotdev/lingo.dev/pull/997#issuecomment-3033030314).

I can still see missing whitespace when the app is rendered:

Screenshot 2025-07-08 at 11 28 36

**Comment by VAIBHAVSING** [comment]
@mathio 
i updated [next.js app demo](https://github.com/lingodotdev/lingo.dev/tree/main/demo/next-app)
this is output
Screenshot from 2025-07-09 16-57-04
Screenshot from 2025-07-09 16-57-13

**Comment by VAIBHAVSING** [comment]
@mathio i think you are on wrong branch i tested for main it wont work 
for jsx-whitespace it work
`git checkout jsx-whitespace`

**Review by mathio — APPROVED** [review]
I probably messed up last time I tested this 🙃 

I rebased your branch on latest `main` and testing again from your fork. It seems to work as expected 🚀 

Screenshot 2025-07-11 at 16 51 54

Thank you for taking the time to contribute! 🙇
